zkvm/src/txlog.rs

```diff
@@ -31,6 +32,27 @@ pub struct TxID(pub [u8; 32]);
 #[derive(Copy, Clone, PartialEq, Debug)]
 pub struct UTXO(pub [u8; 32]);
 
+/// MerkleHash represents a step in a Merkle proof of inclusion.
+#[derive(Copy, Clone, PartialEq, Debug)]
+pub enum MerkleHash {
+    Left([u8; 32]),
+    Right([u8; 32]),
+}
+
+/// MerkleRoot defines the root of a Merkle tree.
+pub struct MerkleRoot {
+    root: MerkleNode,
+    size: usize,
+}
+
+/// MerkleNode represents a precomputed node in the Merkle tree
+/// for easily computing proofs of inclusion.
+struct MerkleNode {
+    left: Option<Box<MerkleNode>>,
+    right: Option<Box<MerkleNode>>,
+    node: [u8; 32],
```

Comment:
Maybe "MerkleNode.hash" sounds better?

```suggestion
    hash: [u8; 32],
```